mithril-test-lab/mithril-end-to-end/src/end_to_end_spec.rs

```diff
@@ -43,224 +50,269 @@ impl<'a> Spec<'a> {
         }
     }
 
-    pub async fn run(&mut self) -> StdResult<()> {
-        let aggregator_endpoint = self.infrastructure.aggregator().endpoint();
-        assertions::wait_for_enough_immutable(self.infrastructure.aggregator().db_directory())
-            .await?;
-        let start_epoch = self
-            .infrastructure
-            .chain_observer()
-            .get_current_epoch()
-            .await?
-            .unwrap_or_default();
+    pub async fn run(self) -> StdResult<()> {
+        let mut join_set = JoinSet::new();
+        let spec = Arc::new(self);
+        let infrastructure_guard = spec.infrastructure.read().await;
+        let infrastructure = infrastructure_guard
+            .as_ref()
+            .ok_or(anyhow!("No infrastructure found"))?;
+        let aggregators = infrastructure_guard
+            .as_ref()
+            .ok_or(anyhow!("No infrastructure found"))?
+            .aggregators();
 
         // Transfer some funds on the devnet to have some Cardano transactions to sign.
         // This step needs to be executed early in the process so that the transactions are available
         // for signing in the penultimate immutable chunk before the end of the test.
         // As we get closer to the tip of the chain when signing, we'll be able to relax this constraint.
-        assertions::transfer_funds(self.infrastructure.devnet()).await?;
+        assertions::transfer_funds(infrastructure.devnet()).await?;
+
+        for index in 0..aggregators.len() {
+            let spec_clone = spec.clone();
+            join_set.spawn(async move {
+                let infrastructure_guard = spec_clone.infrastructure.read().await;
+                let infrastructure = infrastructure_guard
+                    .as_ref()
+                    .ok_or(anyhow!("No infrastructure found"))?;
+
+                spec_clone
+                    .start_aggregator(
+                        infrastructure.aggregator(index),
+                        infrastructure.chain_observer(index),
+                        infrastructure,
+                    )
+                    .await
+            });
+        }
+
+        while let Some(res) = join_set.join_next().await {
+            res??;
+        }
+
+        Ok(())
+    }
+
+    pub async fn start_aggregator(
```

Comment:
This name doesn't fill right, the aggregators are already started (as they start alongside the infrastructure) and this method contains what was before in the main "run".
```suggestion
    pub async fn run_scenario(
```

Reply:
You're right, I updated the name with your suggestion.